src/dmd/dcast.d

```diff
@@ -2801,10 +2842,10 @@ bool typeMerge(Scope* sc, TOK op, Type* pt, Expression* pe1, Expression* pe2)
         }
     }
 
-    Type t1 = e1.type;
-    Type t2 = e2.type;
+    t1 = e1.type;
+    t2 = e2.type;
     assert(t1);
-    Type t = t1;
```

Comment:
This should be moved to its declaration.

Reply:
same.